src/c/types.cpp

```diff
@@ -646,7 +651,7 @@ const int OCCA_PROPERTIES    = occa::c::typeType::properties;
 const occaType occaUndefined  = occa::c::undefinedOccaType();
 const occaType occaDefault    = occa::c::defaultOccaType();
 const occaType occaNull       = occa::c::nullOccaType();
-const occaUDim_t occaAllBytes = -1;
+const occaUDim_t occaAllBytes = static_cast<occaUDim_t>(-1);
```

Comment:
Is this really necessary? I would hope the compiler can figure this out on its own, at least for such primitive types.

Reply:
Clang reports it as an implicit conversion, so yes, it's needed to fix the warning.

Reply:
OK. I just think this makes the code look ugly and cluttered. IMHO, code should be easy to read and intuitive, this is far from that. I would rather consider to add an exception (i.e. suppress the warning) for Clang than adding something like this to the code. Just my two cents...

Reply:
How about we define a const value once and reuse it?
```cpp
const occa::udim_t UDIM_DEFAULT = static_cast<occa::udim_t>(-1);
```

This would also make the code a bit clearer in places where `-1` is set as the default because of how high the number is

Reply:
@dmed256 that's probably a sensible compromise. I like the simple `-1` since it's rather intuitive 
 and easy to understand as is, but having a predefined constant works as well - anything that avoids cluttering the source code with type casts.

Reply:
I've redone the relevant commit to use the const UDIM_DEFAULT, as suggested.